WSC.js

```diff
@@ -827,8 +832,27 @@ _.extend(DirectoryEntryHandler.prototype, {
                                     if (validFile) {
                                         var req = this.request
                                         var res = this
+                                        var clearModuleCache = function() {
+                                            for (var i=0; i<global.cachedFiles.length; i++) {
+                                                delete require.cache[require.resolve(global.cachedFiles[i])]
+                                            }
+                                            global.cachedFiles = [ ]
+                                        }
+                                        var requireFile = function(path) {
+                                            var path = res.fs.mainPath + WSC.utils.relativePath(path, WSC.utils.stripOffFile(res.request.origpath))
+                                            var alreadyInThere = false
+                                            for (var i=0; i<global.cachedFiles.length; i++) {
+                                                if (global.cachedFiles[i] == path) {
+                                                    var alreadyInThere = true
+                                                }
+                                            }
+                                            if (! alreadyInThere) {
+                                                global.cachedFiles.push(path)
+                                            }
+                                            return require(path)
+                                        }
                                         try {
-                                            eval('(function() {var handler = function(req, res, tempData, httpRequest, appInfo) {' + dataa + '};handler(req, res, { }, WSC.httpRequest, {"server": "Simple Web Server"})})();')
+                                            eval('(function() {var handler = function(req, res, tempData, httpRequest, appInfo, clearModuleCache) {' + dataa + '};handler(req, res, { }, WSC.httpRequest, {"server": "Simple Web Server"}, clearModuleCache)})();')
```

Comment:
What does this do?

Reply:
This is the server side POST function

Reply:
Is the user sending javascript that's getting executed?

Reply:
No, as I said [here](https://github.com/terreng/simple-web-server/issues/1#issuecomment-885101664) After checking the filename and the keys the file is excecated. I've blocked every way a hacker could access the key of attempt to upload over the file or attempt to delete the file.

Reply:
So, what script is being executed? I'm actually not familiar with what this feature is. Could you take a few steps back and explain how this feature works for me?

Reply:
The user makes some code in some file using the `howTo/post.md` readme. The user then adds the filename and a random key they typed key to a `wsc.htaccess` file and then server side javascript will work toward the file.

>So, what script is being executed?

The local file the user wrote. Fully server side

Reply:
So, is this basically a way of creating a dynamic server, instead of just serving static files?

Can you give me an example of a use-case for this feature?

Reply:
An example would be something like a file hosting platform, this file could verify and set `Cookie` headers.

I myself made a youtube downloader, where it gets the links and downloads the video to the server

Reply:
I don't think "post" is a very descriptive name for this feature. How about renaming this feature to "custom scripts"?

Reply:
@terreng Just pushed my changes. Can we merge now, this fixes a critical fs bug